test/interface_c.cpp

```diff
@@ -0,0 +1,182 @@
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long"
+                           "this_string_is_just_way_toooooooooooooooooooo_long";
+    EXPECT_DEATH( trixi_initialize( garbage, NULL ),
+                  "buffer size not sufficient for activation command");
+}
+
+
+TEST(CInterfaceTest, VersionInfo) {
+
+    // Initialize Trixi
+    trixi_initialize( julia_project_path, NULL );
+
+    // Check libtrixi version information
+    int major = trixi_version_library_major();
+    int minor = trixi_version_library_minor();
+    int patch = trixi_version_library_patch();
+    std::string version_string("");
+    version_string.append(std::to_string(major));
+    version_string.append(".");
+    version_string.append(std::to_string(minor));
+    version_string.append(".");
+    version_string.append(std::to_string(patch));
+    
+    EXPECT_STREQ(version_string.c_str(), trixi_version_library());
+
+    // Check julia packages version information
+    std::string version_string_julia(trixi_version_julia());
+    EXPECT_NE(version_string_julia.find("OrdinaryDiffEq"), std::string::npos);
+
+    std::string version_string_julia_ext(trixi_version_julia_extended());
+    EXPECT_NE(version_string_julia_ext.find("StartUpDG"), std::string::npos);
+
+    // Finalize Trixi
+    trixi_initialize( julia_project_path, NULL );
+}
+
+
+TEST(CInterfaceTest, JuliaCode) {
+
+    // Initialize Trixi
+    trixi_initialize( julia_project_path, NULL );
+
+    // Execute correct Julia code
+    // NOTE: capturing stdout somehow does not work
+    trixi_eval_julia("println(\"Hello from Julia!\")");
+
+    // Execute erroneous Julia code
+    // NOTE: output before exit is somehow not captured here
+    EXPECT_DEATH(trixi_eval_julia("printline(\"Hello from Julia!\")"), "");
```

Comment:
This was unexpected. 

Calling `trixi_eval_julia` with a wrong julia command results in error messages as expected, see
https://github.com/trixi-framework/libtrixi/blob/5dd0bbdef0428656c390c6b5ebb26d4e47c1f13f/src/auxiliary.c#L81-L92

However this output somehow cannot be captured using gtest's `testing::internal::CaptureStdout()/CaptureStderr()` (which is internal and not documented anyway)

Moreover, in the following, `jl_atexit_hook(1)` results in an error.

> atexit hook threw an error: ReadOnlyMemoryError()

which I do not understand. `exit(1)` is never reached.

In the end it does not really matter. The test fails as expected, but with unexpected last words.

Reply:
Phew. Good question. I don't know. If it works for all non-abnormal cases, let's just be happy about it and ignore this for now :-)

Reply:
Great!